python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -538,3 +553,194 @@ def applySkyCorr(self, dataRef, calexp):
+            inputTypeDict.pop("jointcal_wcs", None)
+            inputTypeDict.pop("jointcal_photoCalib", None)
+        return inputTypeDict
+
+    @classmethod
+    def getOutputDatasetTypes(cls, config):
+        """Return output dataset type descriptors
+
+        Remove output dataset types not produced by the Task
+        """
+        outputTypeDict = super().getOutputDatasetTypes(config)
+        if not config.makeDirect:
+            outputTypeDict.pop("direct", None)
+        if not config.makePsfMatched:
+            outputTypeDict.pop("psfMatched", None)
+        return outputTypeDict
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        """Construct warps for requested warp type for single epoch
+
+        PipelineTask (Gen3) entry point to warp and optionally PSF-match
+        calexpes. This method is analogous to `runDataRef`, it prepares all
+        the data products to be passed to `run`.
+        Return a Struct with only requested warpTypes controlled by the configs
+        makePsfMatched and makeDirect.
+
+        Parameters
+        ----------
+        inputData : `dict`
+            Keys are the names of the configs describing input dataset types.
+            Values are input Python-domain data objects (or lists of objects)
+            retrieved from data butler.
+        inputDataIds : `dict`
+            Keys are the names of the configs describing input dataset types.
+            Values are DataIds (or lists of DataIds) that task consumes for
+            corresponding dataset type.
+        outputDataIds : `dict`
+            Keys are the names of the configs describing input dataset types.
+            Values are DataIds (or lists of DataIds) that task is to produce
+            for corresponding dataset type.
+        butler : `lsst.daf.butler.Butler`
+            Gen3 Butler object for fetching additional data products before
+            running the Task
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+           Result struct with components:
+
+           - ``direct`` : (optional) direct Warp Exposure
+                          (``lsst.afw.image.Exposure``)
+           - ``psfMatched``: (optional) PSF-Matched Warp Exposure
+                            (``lsst.afw.image.Exposure``)
+        """
+        # Construct skyInfo expected by `run`
+        skyMap = inputData["skyMap"]
+        outputDataId = next(iter(outputDataIds.values()))
+        inputData['skyInfo'] = makeSkyInfo(skyMap,
+                                           tractId=outputDataId['tract'],
+                                           patchId=outputDataId['patch'])
+
+        # Construct list of DataId's expected by `run`
+        dataIdList = inputDataIds['calExpList']
+        inputData['dataIdList'] = dataIdList
+
+        # Construct list of ccdIds expected by `run`
+        inputData['ccdIdList'] = [dataId['detector'] for dataId in dataIdList]
```

Comment:
These CCD IDs are not the same as the ones currently being produced by `MakeCoaddTempExp.runDataRef` - those are `ccdExposureIds` that have the visit mangled in.  If you want those here, you can do:
```
inputData["ccdIdList"] = [butler.registry.packDataId("VisitDetector", dataId) for dataId in dataIdList]
```
I am also happy to do this on DM-15843, in which I'm already doing similar things to existing `PipelineTasks` to pass the time between reviews (it's definitely lower-priority than this ticket).

Reply:
Your recipe perfectly replicates the old `ccdExposureIds`. What's the "this" that you're offering?